core/src/saros/activities/FileActivity.java

```diff
@@ -57,7 +61,8 @@ public FileActivity(
       SPath newPath,
       SPath oldPath,
       byte[] content,
-      String encoding) {
+      String encoding,
+      String fileType) {
```

Comment:
nit: Nothing new but we probably should replace this CTOR with static factory methods. Depending on the `FileActivity.Type` different parameters are valid.